File: Linguagens de Programação/GOLang/exponent.go
package main
import ("fmt")

func main() {
    fmt.Println("Calculadora de Expoente")
    var n int 
    var x int
	fmt.Println("Digite o número de Base: ")
    fmt.Scanln(&n)
	fmt.Println("Digite o expoente desejado: ")
	fmt.Scanln(&x)
    var result int = 1
    for i := 0; i < x; i++ {
        result *= n
    }
    fmt.Println("O resultado é: ", result)
}

File: Linguagens de Programação/Haskell/exponent.hs
main = do
    putStrLn "Digite o valor de x:"
    x <- readLn
    putStrLn "Digite o valor de n (não-negativo):"
    n <- readLn
    let result = x ^ n
    putStrLn ("O resultado de " ++ show x ++ "^" ++ show n ++ " é: " ++ show result)
